libnavigation-ui/api/current.txt

```diff
@@ -408,7 +408,7 @@ package com.mapbox.navigation.ui.instruction.maneuver {
     method public void setDrivingSide(String!);
     method public void setManeuverTypeAndModifier(String, String?);
     method public void setPrimaryColor(@ColorInt int);
-    method public void setRoundaboutAngle(@FloatRange(from=60.0f, to=300.0f) float);
```

Comment:
This is breaking SemVer, is that expected?

Reply:
we still can set `float`, it'll be casted to `Float`. No breaking changes

Reply:
For Kotlin users is a breaking change as it's `@Nullable` now.

Reply:
they still can set `NonNull` values, otherwise it would be a problem, doesn't it?

Reply:
The problem is that primitives are `@NonNull` and if we change it to `@Nullable` 💥

Reply:
> For Kotlin users is a breaking change as it's @Nullable now.

I've been thinking about it as well, and does it? We are relaxing the constraints, it shouldn't break semver. It's an argument type, not a return type.

Reply:
That's a good point. Isn't Metalava taking that into account? 🤔

Reply:
Metalava just reports the status (which does change in this case), it's not flagging that something is breaking, is it?

Reply:
I would say so. It's only checking public API changes

Reply:
Yeah, I'd say so. From https://android.googlesource.com/platform/tools/metalava/

> Like doclava1, metalava can diff two APIs and warn about API compatibility problems such as removing API elements. Metalava adds new warnings around nullness, such as attempting to change a nullness contract incompatibly (e.g. you can change a parameter from non null to nullable for final classes, but not versa).

Reply:
OK. Let's

> accept neatness as a disease

and see what happens 😅